source/menu/menu_coop.qc

```diff
@@ -0,0 +1,199 @@
+
+float player_name_cursor;
+string player_name;
+
+float server_ip_cursor;
+string server_ip;
+
+float server_hostname_cursor;
+string server_hostname;
+
+float server_password_cursor;
+string server_password;
+
+float server_port_cursor;
+string server_port;
+
+int num_cached_servers;
+
+void() Menu_Coop_Name_Update =
+{
+	static float initialized;
+	if (!initialized)
+	{
+		player_name = cvar_string("name");
+		player_name_cursor = strlen(player_name);
+		initialized = true;
+	}
+	else
+	{
+		cvar_set("name", player_name);
+	}
+};
+
+void() Menu_Coop_Hostname_Update =
+{
+	static float initialized;
+	if (!initialized)
+	{
+		server_hostname = cvar_string("hostname");
+		server_hostname_cursor = strlen(server_hostname);
+		initialized = true;
+	}
+	else
+	{
+		cvar_set("hostname", server_hostname);
+	}
+};
+
+void() Menu_Coop_Port_Update =
+{
+	static float initialized;
+	if (!initialized)
+	{
+		server_port = cvar_string("sv_port");
+		server_port_cursor = strlen(server_port);
+		initialized = true;
+	}
+	else
+	{
+		cvar_set("sv_port", server_port);
+	}
+};
+
+void() Menu_Coop_Password_Update =
+{
+	static float initialized;
+	if (!initialized)
+	{
+		server_password = cvar_string("password");
+		server_password_cursor = strlen(server_password);
+		initialized = true;
+	}
+	else
+	{
+		cvar_set("password", server_password);
+	}
+};
+
+void() Menu_Coop_Browse_Refresh =
+{
+	static float refreshtime;
+
+	if (refreshtime > time)
+		return;
+
+	refreshtime = time + 0.5;
```

Comment:
Can we speed this up? Building the cache list is very slow, setting `com_protocolname` to `FTE-Quake` demonstrates this

Reply:
Possibly resolved in d6c3934b73dc29f30dae81b646ffd5449dbd12cb.